File: lib-base/src/main/java/com/llj/lib/base/IBaseActivity.java
package com.llj.lib.base;

import android.app.Activity;
import android.arch.lifecycle.Lifecycle;
import android.support.annotation.NonNull;
import android.view.MotionEvent;

import com.llj.lib.utils.AInputMethodManagerUtils;

/**
 * ArchitectureDemo
 * describe:
 * author liulj
 * date 2018/5/24
 */
public interface IBaseActivity extends IActivityStack {

    void initLifecycleObserver(@NonNull Lifecycle lifecycle);

    void superOnBackPressed();

    void backToLauncher(boolean nonRoot);

    default boolean onTouchEvent(Activity activity, MotionEvent event) {
        if (event.getAction() == MotionEvent.ACTION_DOWN) {
            AInputMethodManagerUtils.hideSoftInputFromWindow(activity);
        }
        return activity.onTouchEvent(event);
    }
}

File: lib-base/src/main/java/com/llj/lib/base/IBaseActivity.kt
package com.llj.lib.base

import android.app.Activity
import android.graphics.Insets
import android.view.*
import androidx.lifecycle.Lifecycle
import com.llj.lib.statusbar.LightStatusBarCompat
import com.llj.lib.statusbar.StatusBarCompat
import com.llj.lib.utils.AInputMethodManagerUtils

/**
 * ArchitectureDemo
 * describe:
 * @author llj
 * @date 2018/5/24
 */
interface IBaseActivity : IActivityStack, IModuleName {

  fun initLifecycleObserver(lifecycle: Lifecycle)

  fun superOnBackPressed()

  fun backToLauncher(nonRoot: Boolean)

  fun onTouchEvent(activity: Activity, event: MotionEvent) {
    if (event.action == MotionEvent.ACTION_DOWN) {
      AInputMethodManagerUtils.hideSoftInputFromWindow(activity)
    }
  }

}
